openapi.yaml

```diff
@@ -0,0 +1,2412 @@
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  content:
+                    type: string
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+        '404':
+          $ref: '#/components/responses/NotFound'
+
+  /config:
+    get:
+      summary: Get current configuration
+      operationId: getConfig
+      tags: [Config]
+      parameters:
+        - $ref: '#/components/parameters/PageLimit'
+        - $ref: '#/components/parameters/PageOffset'
+      responses:
+        '200':
+          description: Return current configuration.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Config'
+            text/plain:
+              schema:
+                type: string
+        '401':
+          $ref: '#/components/responses/Unauthenticated'
+        '403':
+          $ref: '#/components/responses/PermissionDenied'
+
+components:
+  # Reusable schemas (data models)
+  schemas:
+    # Database entities
+    ConnectionCollectionItem:
+      type: object
+      properties:
+        connection_id:
+          type: string
+        conn_type:
+          type: string
+        host:
+          type: string
+        login:
+          type: string
+        schema:
+          type: string
+        port:
+          type: integer
+
+    ConnectionCollection:
+      type: object
+      properties:
+        connections:
+          type: array
+          items:
+            $ref: '#/components/schemas/ConnectionCollectionItem'
+
+    Connection:
+      allOf:
+        - $ref: '#/components/schemas/ConnectionCollectionItem'
+        - type: object
+          properties:
+            password:
+              type: string
+              format: password
+              writeOnly: true
+            extra:
+              type: string
+
+    DAG:
+      type: object
+      properties:
+        dag_id:
+          type: string
+          readOnly: true
+        root_dag_id:
+          type: string
+          readOnly: true
+        is_paused:
+          type: boolean
+        is_subdag:
+          type: boolean
+          readOnly: true
+        fileloc:
+          type: string
+          readOnly: true
+        file_token:
+          type: string
+          readOnly: true
+          description: >
+            The key containing the encrypted path to the file. Encryption and decryption take place only on
+            the server. This prevents the client from reading an non-DAG file. This also ensures API
+            extensibility, because the format of encrypted data may change.
+        owners:
+          type: array
+          items:
+            type: string
+          readOnly: true
+        description:
+          type: string
+          readOnly: true
+        schedule_interval:
+          oneOf:
```

Comment:
```suggestion
          allOf:
```

As discussed in devlist - this allows to get combilable and testable client APIs

Reply:
Actually we don't need oneOf/all of at all here:

```yaml
         schedule_interval:
           $ref: '#/components/schemas/ScheduleInterval'
           readOnly: true
```

That might work. (It would work with JSON Schema, but the interactions/overlap between JsonSchema and OpenAPI are not always clear

Reply:
Will compile cleanly and give us working clients? Are we sure about it?

Reply:
My IDE is sad. Did I miss something?
<img width="610" alt="Screenshot 2020-06-01 at 14 49 29" src="https://user-images.githubusercontent.com/12058428/83410717-32b2c080-a417-11ea-9668-45b820dc8922.png">
<img width="593" alt="Screenshot 2020-06-01 at 14 50 48" src="https://user-images.githubusercontent.com/12058428/83410767-4d853500-a417-11ea-9964-5a0bdb02a039.png">

Reply:
https://github.com/apache/airflow/pull/8721/commits/31d8e1574fea098f72acf277458a81d7e0789c79

Reply:
if you need to merge attribute (readonly) with predefined schema in openapi, you will need to use allOf unfortunately :(

Reply:
Ah k, so yeah, this was my knowledge of JSONSchema biting me here cos this "looks" like JSONSchema but isn't :(